keras_cv/models/segmentation/fcn_test.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pytest
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+from keras_cv import models
+from keras_cv.models.segmentation import FCN
+
+
+class FCNTest(tf.test.TestCase):
+    def test_fcn_model_with_vgg16_backbone_construction_with_preconfigured_setting(
+        self,
+    ):
+        model = FCN(
+            classes=11,
+            backbone="vgg16",
+            model_architecture="fcn8s",
+            input_shape=(256, 256, 3),
+        )
+        input_image = tf.random.uniform(shape=[2, 256, 256, 3])
+        output = model(input_image)
+
+        self.assertEquals(output.shape, [2, 256, 256, 11])
+
+    def test_fcn_model_with_vgg19_backbone_construction_with_preconfigured_setting(
+        self,
+    ):
+        model = FCN(
+            classes=11,
+            backbone="vgg19",
+            model_architecture="fcn8s",
+            input_shape=(256, 256, 3),
+        )
+        input_image = tf.random.uniform(shape=[2, 256, 256, 3])
+        output = model(input_image)
+
+        self.assertEquals(output.shape, [2, 256, 256, 11])
+
+    def test_fcn_model_with_vgg16_components(self):
+        backbone = models.VGG16(include_rescaling=False, include_top=False)
+        print(type(backbone))
+        model = FCN(classes=11, backbone=backbone, input_shape=(256, 256, 3))
```

Comment:
Sure, but will require setting the channel to 1 instead of the conventional 3. I think it's mostly dependent on whether the backbone can handle it (which I think it can)

If setting to HxW only, I haven't tested that yet

Reply:
Since the input shape should be inherited from the backbone - as long as the backbone is set up, it should work.
For reference:

```
    def test_greyscale_input(self):
        backbone = models.ResNet50V2(
            include_rescaling=True, include_top=False, input_shape=[64, 64, 1]
        )
        model = segmentation.DeepLabV3(classes=11, backbone=backbone)
        input_image = tf.random.uniform(shape=[1, 64, 64, 1])
        output = model(input_image, training=True)

        self.assertEquals(output["output"].shape, [1, 64, 64, 11])
```

Reply:
Yeah, I think then it should pass. We can always revisit this test later on.